src/axiselements.jl

```diff
@@ -310,83 +308,214 @@ function print_tex(io::IO, coordinates::Coordinates)
+    colnames = ["x", "y", "z"]
+    columns = hcat(matrix_xyz(x, y, z)...)
+    if meta ≠ nothing
+        @argcheck size(z) == size(meta) "Incompatible sizes."
+        push!(colnames, "meta")
+        columns = hcat(columns, vec(meta))
+    end
+    columns, colnames, length(x)
 end
 
-function Table(args::Vararg{PGFOption}; kwargs...)
-    Table(kwargs, dictify(args))
+"""
+    Table([options], args...)
+
+Construct a table from the given arguments. Examples:
+
+```julia
+Table(["x" => 1:10, "y" => 11:20])        # from a vector
+
+Table([1:10, 11:20])                      # same contents, unnamed
+
+Table(Dict(:x => 1:10, :y = 11:20))       # a Dict with symbols
+
+Table(@pgf { "x index" = 2, "y index" = 1" }, randn(10, 3))
+
+let x = linspace(0, 1, 10), y = linspace(-2, 3, 15)
+    Table(x, y, sin.(x + y'))             # edges & matrix
 end
+```
+
+[`table_fields`](@ref) is used to convert the arguments after options. See its
+methods for possible conversions.
+"""
+Table(options::OrderedDict{Any, Any}, args...; kwargs...) =
+    Table(options, table_fields(args...; kwargs...)...)
+
+Table(args...; kwargs...) =
+    Table(OrderedDict{Any, Any}(), table_fields(args...; kwargs...)...)
 
-function print_tex(io_main::IO, t::Table)
+function print_tex(io_main::IO, table::Table)
+    @unpack options, data, colnames, scanlines = table
     print_indent(io_main) do io
         print(io, "table ")
-        print_options(io, t.options)
+        print_options(io, options)
         print(io, "{")
-        print_tex(io, t.data, t)
+        _sep() = print(io, "  ")
+        if colnames ≠ nothing
+            for colname in colnames
+                print(io, colname)
+                _sep()
+            end
+        end
+        println(io)
+        for row_index in indices(data, 1)
+            for col_index in indices(data, 2)
+                print_tex(io, data[row_index, col_index])
+                _sep()
+            end
+            println(io)
+            if row_index ∈ scanlines
+                println(io)
+            end
+        end
         print(io, "}")
     end
 end
 
-print_tex(io::IO, str::String, ::Table) = print(io, str)
-function print_tex(io::IO, v::AbstractVector, ::Table)
-    length(v) == 0 && return
-    if v[1] isa Tuple # Assume the kw constructor was called
-        # Do some basic checking
-        # Print header
-        vs = []
-        first = true
-        l = -1
-        for s in v
-            if !(length(s) == 2) || !(s[1] isa Symbol) || !(s[2] isa AbstractVector)
-                error("Expected a call like Table(; a = [1,2,...], b = [2,3,...])")
-            end
-
-            if first
-                l = length(s[2])
-            end
+#############
+# TableFile #
+#############
 
-            if !(l == length(s[2]))
-                error("length of data in columns not the same")
-            end
-
-            print(io, s[1], "    ")
-            push!(vs, s[2])
-        end
+struct TableFile <: OptionType
```

Comment:
What's the reason over just e.g. `Table(path)`.

Reply:
You mean `Table(path)` constructing a `TableFile`, or simply storing the `path` in a `Table`?

I have in mind an interface for writing `Table` data into files and returning an appropriate `TableFile`, ie externalizing tables, for larger datasets, but plan to explore this later.

Reply:
I am now convinced that it is best if we keep very close to the syntax of `pgfplots`, so I modified code to use the `Table` constructor for these. Thanks!